src/main/java/teammates/ui/pagedata/InstructorCoursesPageData.java

```diff
@@ -166,8 +176,49 @@ private ActiveCoursesTable convertToActiveCoursesTable(List<CourseAttributes> co
         return activeCourses;
     }
 
+    private RecoveryCoursesTable convertToRecoveryCoursesTable(List<CourseAttributes> courses) {
+        RecoveryCoursesTable recoveryCourses = new RecoveryCoursesTable();
+
+        int idx = -1;
+
+        for (CourseAttributes course : courses) {
+            idx++;
+
+            List<ElementTag> actionsParam = new ArrayList<>();
+
+            String restoreLink = getInstructorCourseRestoreRecoveryCourseLink(course.getId());
+            ElementTag restoreButton = createButton("Restore", "btn btn-default btn-xs t_course_restore" + idx, "",
+                    restoreLink, Const.Tooltips.COURSE_RESTORE, false, "");
+
+            String deleteLink = getInstructorCourseDeleteRecoveryCourseLink(course.getId(), false);
+            Boolean hasDeletePermission = instructorsForCourses.get(course.getId()).isAllowedForPrivilege(
+                    Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE);
+            ElementTag deleteButton = createButton("Delete Permanently", "btn btn-default btn-xs course-delete-link "
+                            + "t_course_delete" + idx, "", deleteLink, Const.Tooltips.COURSE_DELETE,
+                    !hasDeletePermission, "color: red");
```

Comment:
In this case since the "style" part is only used once (here), I am not sure if it would actually be cleaner to use the original `createButton` method and then just add a `deleteButton.setAttribute` line. Hmm....

Reply:
Yeah I agree. Also, for now since I have to `createButton` method in both `InstructorCoursesPageData` and `InstructorFeedbackSessionsPageData`, is it better if I move the method to `PageData`?